src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageRouteModel.cs

```diff
@@ -52,6 +46,7 @@ public PageRouteModel(PageRouteModel other)
 
             Properties = new Dictionary<object, object>(other.Properties);
             Selectors = new List<SelectorModel>(other.Selectors.Select(m => new SelectorModel(m)));
+            RouteValues = new Dictionary<string, string>(other.RouteValues, StringComparer.OrdinalIgnoreCase);
```

Comment:
So in theory we probably should have had this here always right?

Reply:
Yup. We missed a couple of features in PageApplicatonModel \ PageRouteModel that controllers had.